System/Writer.h

```diff
@@ -80,49 +80,57 @@ namespace RTE {
 		/// Used to specify the start of an object to be written.
 		/// </summary>
 		/// <param name="className">The class name of the object about to be written.</param>
-		virtual void ObjectStart(const std::string &className) { *m_pStream << className; ++m_Indent; }
+		virtual void ObjectStart(const std::string &className) { *m_Stream << className; ++m_Indent; }
 
 		/// <summary>
 		/// Used to specify the end of an object that has just been written.
 		/// </summary>
-		virtual void ObjectEnd() {
-			--m_Indent;
-			// Make an extra line between big object definitions
-			if (m_Indent == 0) { *m_pStream << "\n\n"; }
-		}
+		virtual void ObjectEnd() { --m_Indent; if (m_Indent == 0) { NewLine(false, 2); } }
+
+		/// <summary>
+		/// Creates a new line that can be properly indented.
+		/// </summary>
+		/// <param name="toIndent">Whether to indent the new line or not.</param>
+		/// <param name="lineCount">How many new lines to create.</param>
+		virtual void NewLine(bool toIndent = true, unsigned short lineCount = 1);
 
 		/// <summary>
-		/// Creates a new line that is properly indented.
+		/// Creates a new line and writes the specified string to it.
 		/// </summary>
-		virtual void NewLine() {
-			*m_pStream << "\n";
-			for (int i = 0; i < m_Indent; ++i) {
-				*m_pStream << "\t";
+		/// <param name="textString">The text string to write to the new line.</param>
+		/// <param name="toIndent">Whether to indent the new line or not.</param>
+		virtual void NewLineString(std::string textString, bool toIndent = true) { NewLine(toIndent); *m_Stream << textString; }
+
+		/// <summary>
+		/// Creates a new line and fills it with slashes to create a divider line for INI.
+		/// </summary>
+		/// <param name="toIndent">Whether to indent the new line or not.</param>
+		/// <param name="dividerLength">The length of the divider (number of slashes).</param>
+		virtual void NewDivider(bool toIndent = true, unsigned short dividerLength = 72) {
```

Comment:
When I did the Data refactor I made all the ini dividers 72 slashes long because I wasn't counting or thinking about ever needing that count. ¯\\\_(ツ)_/¯

Reply:
:cri: fair enough, it is what it is.